spec/integration/defaults_spec.rb

```diff
@@ -28,6 +28,19 @@
       Puppet.settings[:masterport] = 3939
       expect(Puppet.settings[:serverport]).to eq(3939)
     end
+
+    it "should not overwrite :serverport if explicitly set" do
+      Puppet.settings[:serverport] = 9000
+      Puppet.settings[:masterport] = 9001
+      expect(Puppet.settings[:serverport]).to eq(9000)
+    end
+  end
+
+  describe "when setting the :serverport" do
+    it "should also set the :masterport to the same value" do
+      Puppet.settings[:serverport] = 9000
+      expect(Puppet.settings[:masterport].to_i).to eq(9000)
```

Comment:
The `to_i` coercion has to be done due to https://github.com/puppetlabs/puppet/blob/601abf8/lib/puppet/defaults.rb#L1363...